src/main/java/com/esri/samples/na/service_areas/ServiceAreaTaskController.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright 2016 Esri.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.esri.samples.na.service_areas;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javafx.fxml.FXML;
+import javafx.geometry.Point2D;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Alert.AlertType;
+import javafx.scene.control.Button;
+import javafx.scene.control.ToggleButton;
+import javafx.scene.input.MouseButton;
+
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.Polyline;
+import com.esri.arcgisruntime.geometry.PolylineBuilder;
+import com.esri.arcgisruntime.geometry.SpatialReference;
+import com.esri.arcgisruntime.geometry.SpatialReferences;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.symbology.PictureMarkerSymbol;
+import com.esri.arcgisruntime.symbology.SimpleFillSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.networkanalysis.PolylineBarrier;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaFacility;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaParameters;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaPolygon;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaResult;
+import com.esri.arcgisruntime.tasks.networkanalysis.ServiceAreaTask;
+
+public class ServiceAreaTaskController {
+
+  @FXML private MapView mapView;
+  @FXML private ToggleButton btnAddFacility;
+  @FXML private ToggleButton btnAddBarrier;
+  @FXML private Button btnShowServiceArea;

```

Comment:
These last two Buttons are not used in the code. They don't need to be listed here.